src/annotator/components/test/NotebookModal-test.js

```diff
@@ -0,0 +1,135 @@
+import NotebookModal from '../NotebookModal';
+import { EventBus } from '../../util/emitter';
+import { mount } from 'enzyme';
+import { act } from 'preact/test-utils';
+
+describe('NotebookModal', () => {
+  let container;
+  let components;
+  let eventBus;
+
+  const createComponent = config => {
+    const component = mount(
+      <NotebookModal
+        eventBus={eventBus}
+        config={{ notebookAppUrl: '/notebook', ...config }}
+      />,
+      {
+        attachTo: container,
+      }
+    );
+    components.push(component);
+    return component;
+  };
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    components = [];
+    document.body.appendChild(container);
+    eventBus = new EventBus();
+  });
+
+  afterEach(() => {
+    components.forEach(component => component.unmount());
+    container.remove();
+  });
+
+  it('hides modal on first render', () => {
+    const wrapper = createComponent();
+    const outer = wrapper.find('.Notebook__outer');
+
+    assert.isTrue(outer.hasClass('is-hidden'));
+    assert.equal(getComputedStyle(wrapper.getDOMNode()).display, 'none');
+    assert.isFalse(wrapper.find('iframe').exists());
+  });
+
+  it('shows modal on "openNotebook" event', () => {
+    const wrapper = createComponent();
+    let outer = wrapper.find('.Notebook__outer');
+
+    assert.isTrue(outer.hasClass('is-hidden'));
+    assert.equal(getComputedStyle(wrapper.getDOMNode()).display, 'none');
+    assert.isFalse(wrapper.find('iframe').exists());
+
+    eventBus.createEmitter().publish('openNotebook', 'myGroup');
```

Comment:
It would probably make sense for `EventBus` to have a method for publishing events directly rather than needing to create an emitter each time.

Reply:
Good idea.